src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Unix.cs

```diff
@@ -10,47 +10,77 @@
 using System.Runtime.InteropServices;
 using System.Threading;
 using Microsoft.DotNet.XUnitExtensions;
+using Microsoft.DotNet.RemoteExecutor;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace System.IO.Tests
 {
-    public partial class DangerousFileSystemWatcherTests
+    // Helper class to avoid issues with RemoteExecutor and _output and other properties.
+    internal class RunTestHelper
     {
-        [ConditionalFact]
-        [OuterLoop("Slow test with significant resource usage.")]
-        public void FileSystemWatcher_Unix_DoesNotLeak()
+        private int _count;
+        private int _limit;
+        private string _path;
+
+        internal RunTestHelper(int count, int limit, string path)
         {
-            Interop.Sys.GetRLimit(Interop.Sys.RlimitResources.RLIMIT_NOFILE, out Interop.Sys.RLimit limits);
-            _output.WriteLine("File descriptor limit is {0}", limits.CurrentLimit);
+            _count = count;
+            _limit = limit;
+            _path = path;
+        }
 
-            if (limits.CurrentLimit > 50_000)
+        internal void ExecuteTest()
+        {
+            RemoteInvokeOptions options = new RemoteInvokeOptions{ TimeOut = 600_000 };
+            RemoteExecutor.Invoke((countString, openFileLimitString, testDirectory) =>
             {
-                throw new SkipTestException($"File descriptor limit is too high {limits.CurrentLimit}.");
-            }
+                ulong maxFd = ulong.Parse(openFileLimitString);
+                int count = Int32.Parse(countString);
+                Interop.Sys.RLimit limits = new Interop.Sys.RLimit{ CurrentLimit = maxFd, MaximumLimit = maxFd};
+
+                // Set open file limit to given value.
+                Assert.Equal(0, Interop.Sys.SetRLimit(Interop.Sys.RlimitResources.RLIMIT_NOFILE, ref limits));
+                Assert.Equal(0, Interop.Sys.GetRLimit(Interop.Sys.RlimitResources.RLIMIT_NOFILE, out limits));
+                Assert.Equal(maxFd, limits.CurrentLimit);
 
-            try {
-                for (int i = 0; i < (int)limits.CurrentLimit; i++)
+                try
                 {
-                    using (var watcher = new FileSystemWatcher(TestDirectory))
+                    for (int i = 0; i < count; i++)
                     {
-                        watcher.Created += (s, e) => { } ;
-                        watcher.EnableRaisingEvents = true;
+                        using (var watcher = new FileSystemWatcher(testDirectory))
+                        {
+                            watcher.Created += (s, e) => { } ;
+                            watcher.EnableRaisingEvents = true;
+                            watcher.EnableRaisingEvents = false;
```

Comment:
Is this necessary?

Reply:
no, it is not. I had issues reproducing the original problem and failing the test. I don't know if it makes sense to test disposal in both cases when the watcher is running or not. But for now, I removed the last assignment to keep it simple.